src/index.js

```diff
@@ -191,10 +191,12 @@ export function initialize(env, user, specifiedOptions, platform, extraDefaults)
     return utils.wrapPromiseCallback(
       clearFirst
         .then(() => userValidator.validateUser(user))
-        .then(realUser => ident.setUser(realUser))
-        .then(() => requestor.fetchFlagSettings(ident.getUser(), hash))
-        .then(requestedFlags => {
+        .then(realUser =>
+          requestor.fetchFlagSettings(realUser, hash).then(requestedFlags => ({ requestedFlags, realUser }))
+        )
+        .then(({ requestedFlags, realUser }) => {
           const flagValueMap = utils.transformVersionedValuesToValues(requestedFlags);
+          ident.setUser(realUser);
```

Comment:
This would of course be easier if we could write it as async code (some day), but I believe you could skip the extra `.then` step by nesting the next one, like this:

```
        .then(realUser =>
          requestor.fetchFlagSettings(realUser, hash)
            .then(requestedFlags => (
              const flagValueMap = utils.transformVersionedValuesToValues(requestedFlags);
              ident.setUser(realUser);
```